### PR TITLE
Updated the setup to support installing to Visual Studio 2015.

### PR DESCRIPTION
**Pull request by billings7**
Partially addresses #18

Full support for 2015 still needs CCI to be updated to support the custom metadata added by the Roslyn compiler when `dynamic` is used.

### CODE REVIEW

**Comment by aarondandy** [comment]
Is anybody watching this PR with access to upload a new installer?

**Comment by VSadov** [comment]
Re: needs CCI to be updated to support the custom metadata added by the Roslyn 

As I understand the most recent version of contract tools should already have updated CCI that has fixes to handle Roslyn outputs. I am not familiar with this codebase though, so I do not know where CCI lives in the tree.

cc @tmat

**Comment by billings7** [comment]
The assemblies in Microsoft.Research\ImportedCCI2 are what needs to be updated. I could do it, however I can't seem to find the source for a couple of the assemblies, AsmMeta.exe an CCDoc.exe.

**Comment by vossad01** [comment]
With VS2015 having released an RC build, it would be great for a build with this to be released :)

**Comment by hubuk** [comment]
Already recompiled Microsoft.Cci.PdbReader.dll with most recent changes from Codeplex. Commit is pending...

**Comment by billings7** [comment]
I'll have to check if an updated CCI will fix it but I haven't been able to build a project I am working on in 2015, the rewriter fails with some error about not being able to resolve a type.

**Comment by hubuk** [comment]
The error message with type resolving is probably related to Roslyn issue regarding compilation of VisualBasic projects with /nostdlib or /noconfig. I will check it later and raise an issue to Roslyn if confirmed.

**Comment by hubuk** [comment]
:+1:  Tested on Visual Studio Enterprise 2015 RC. Seems to be working. Great work!

**Comment by Daniel-Svensson** [comment]
:thumbsup: Installation seems to go fine and code contracts is run as expected under 2015 RC. So a god step on the way to full support for VS 2015

It does seems like it does not rewrite Contracts.Requires(...) inside async methods under VS 2015 (with roslyn) but under VS 2013 it works fine, but that is another problem.

**Comment by hubuk** [comment]
Finally diagnosed what was causing errors about not being able to resolve a type. It was a wrong MyType for Microsoft.VisualBasic.Contracts project. Fixed in #43.
Despite some warnings about metadata version compilation under VS2015 shall be working now (with #39 #40 and #43).

**Comment by billings7** [comment]
The problem I have is caused by another issue. I have a solution, just need to get around to do a pull request.

**Comment by billings7** [comment]
I have fixed one problem I was having (#47, pr #48), but that has lead me on to another problem that I currently have no clue how to solve (#49)

File: Microsoft.Research/ManagedContract.Setup/MSBuild/v14.0/Microsoft.CodeContracts.targets [hunk 1]
```diff
@@ -0,0 +1,654 @@
...
+    <Exec
+       Command='"$(CodeContractsCCdocgenCommand)" "@$(IntermediateOutputPath)$(ProjectName)ccdocgen.rsp"' 
+       />
+  </Target>
+
+  <!--=====================================================================
+      Include Code Analysis target if present
+    ======================================================================-->
+  <PropertyGroup>
+    <CodeContractAnalysisTargets>$(CodeContractsInstallDir)MsBuild\v4.0\Microsoft.CodeContractAnalysis.targets</CodeContractAnalysisTargets>
```
Comment (hubuk) [inline Microsoft.Research/ManagedContract.Setup/MSBuild/v14.0/Microsoft.CodeContracts.targets]: According to issue #30 this line should point to v14.0 of the targets folder.

**Comment by sharwell** [comment]
:thought_balloon: The alternative to this is to distribute the MSBuild tooling exclusively as a NuGet package starting with the Visual Studio 2015 support, and use that as a model for transitioning the older tools as well.

**Comment by SergeyTeplyakov** [comment]
Guys, what is the current status of this PR?

**Comment by billings7** [comment]
@hubuk suggested that the binary files should not be committed, but be created by part of the build process, which I agree with. However I feel that would be better suited for a different PR, so leave them here for now as they would be a pain for anyone else to built ATM (they were for me anyway) and so we can get an updated setup out ASAP. As long as we are fine with that, this one should be good to go.

**Comment by SergeyTeplyakov** [comment]
Merged manually.

**Pull request closed without merging**